data-service/src/main/java/com/amazonaws/blox/dataservice/repository/EnvironmentRepositoryDDB.java

```diff
@@ -383,4 +389,31 @@ public void deleteEnvironmentRevision(@NonNull final EnvironmentRevision environ
           e);
     }
   }
+
+  @Override
+  public List<Cluster> listClusters(String accountId, String clusterNamePrefix) {
+    if (clusterNamePrefix != null) {
+      Validate.notNull(accountId, "accountId must be specified if clusterNamePrefix is specified");
+    }
+
+    // TODO: rewrite this to use the index instead of scanning the table
+    PaginatedScanList<EnvironmentDDBRecord> records =
+        dynamoDBMapper.scan(EnvironmentDDBRecord.class, new DynamoDBScanExpression());
+
+    Stream<Cluster> stream =
+        records
+            .stream()
+            .map(EnvironmentDDBRecord::getAccountIdCluster)
+            .distinct()
+            .map(Cluster::fromAccountIdCluster);
+
+    if (accountId != null) {
+      stream = stream.filter(c -> c.getAccountId().equals(accountId));
```

Comment:
How about chain with the stream?  
```
.map(..)
.filter(c -> accountId==null || c.getAccountId().equals(accountId)))
```

Reply:
I'm keeping this separate so that I can change it over to the DDB Query expression code quickly in the followup PR.